pkg/p2p/message_handler_manager.go

```diff
@@ -0,0 +1,149 @@
+package p2p
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultHandlerOperationTimeout = 3 * time.Second
+)
+
+// MessageHandlerManager is for managing message topic handlers.
+// NOTE: for each topic, only one handler is allowed.
+type MessageHandlerManager interface {
+	RegisterHandler(ctx context.Context, topic Topic, tpi TypeInformation, fn HandlerFunc) (bool, error)
+
+	UnregisterHandler(ctx context.Context, topic Topic) (bool, error)
+
+	CheckError(ctx context.Context) error
+
+	// Clean unregisters all existing handlers.
+	Clean(ctx context.Context) error
+
+	// SetTimeout sets the timeout for handler operations.
+	// A timeout is needed because the underlying handler operations are
+	// asynchronous in the MessageServer.
+	SetTimeout(timeout time.Duration)
```

Comment:
If we expose this function, does it mean we have demand of setting timeout on the fly?

Reply:
- Allowing setting the timeout in the constructor will make the constructor function too complicated.
- Making the timeout a constant will make unit testing difficult.
Although we do not have demand for resetting the timeout on the fly, I think exposing a method is the better way.

Reply:
ok let it be